### PR TITLE
[Audio 4/?] Build Samplebanks, match Audiotable

### PR DESCRIPTION
**Pull request by Thar0**
This PR introduces the Samplebank Compiler `sbc` for building samplebanks. With this, the `Audiotable` segment is fully matched.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -274,7 +291,19 @@ DEP_FILES := $(O_FILES:.o=.asmproc.d) $(OVL_RELOC_FILES:.o=.d)
...
+                     $(ASM_DIRS) \
+                     $(ASSET_BIN_DIRS) \
+                     $(ASSET_BIN_DIRS_C_FILES) \
+                     $(SAMPLE_DIRS) \
+                     $(SAMPLEBANK_DIRS),\
+                   $(BUILD_DIR)/$(dir)) \
+                 $(foreach dir,\
+                     $(SAMPLE_EXTRACT_DIRS) \
+                     $(SAMPLEBANK_EXTRACT_DIRS), \
+                   $(dir:$(EXTRACTED_DIR)/%=$(BUILD_DIR)/%)))
```
Comment (hensldm) [inline Makefile:306]:
```suggestion
$(shell mkdir -p $(foreach dir,\
                     $(SRC_DIRS) \
                     $(ASM_DIRS) \
                     $(OTHER_DIRS),\
                     $(BUILD_DIR)/$(dir)))
$(shell mkdir -p $(foreach dir,\
                     $(ASSET_BIN_DIRS) \
                     $(ASSET_BIN_DIRS_C_FILES) \
                     $(SAMPLE_DIRS) \
                     $(SAMPLEBANK_DIRS),\
                   $(BUILD_DIR)/$(dir)) \
                 $(foreach dir,\
                     $(SAMPLE_EXTRACT_DIRS) \
                     $(SAMPLEBANK_EXTRACT_DIRS),\
                   $(dir:$(EXTRACTED_DIR)/%=$(BUILD_DIR)/%)))
```
Looks MM specific, but at least for me, the `mkdir` argument list has gotten too long
```bash
make: /bin/bash: Argument list too long
```
My suggestion is to break it up into 2 commands. The way I have it up above splits it into roughly src directories in one command and assets in the other.

**Review by hensldm — COMMENTED** [review]
1 MM specific review (though could technically apply to OoT if you wanted). Will add any tooling review on the OoT side.

File: Makefile [hunk 2]
```diff
@@ -548,6 +581,32 @@ ifeq ($(AUDIO_BUILD_DEBUG),1)
...
+.PRECIOUS: $(BUILD_DIR)/assets/audio/samplebanks/%.s
+$(BUILD_DIR)/assets/audio/samplebanks/%.s: $(BUILD_DIR)/assets/audio/samplebanks/%.xml | $(AIFC_FILES) $(SAMPLE_BLOBS)
+	$(SBC) $(SBCFLAGS) --makedepend $(@:.s=.d) $< $@
+
+-include $(SAMPLEBANK_DEP_FILES)
+
+$(BUILD_DIR)/assets/audio/samplebanks/%.o: $(BUILD_DIR)/assets/audio/samplebanks/%.s
+	$(AS) $(ASFLAGS) $< -o $@
+ifeq ($(AUDIO_BUILD_DEBUG),1)
+	$(OBJCOPY) -O binary -j.rodata $@ $(@:.o=.bin)
```
Comment (AngheloAlf) [inline Makefile:606]: Long options are nicer most of the time.
```suggestion
	$(OBJCOPY) -O binary --only-section=.rodata $@ $(@:.o=.bin)
```

File: Makefile (hunk 1 above)
Comment (Thar0) [reply Makefile]: I've tried splitting it into 3 commands, could you check again?

Comment (hensldm) [reply Makefile]: Seems to work for me now. Thank you :)

**Review by hensldm — APPROVED** [review]

**Review by AngheloAlf — APPROVED** [review]

**Pull request merged**